### ISSUE TITLE
Extracting e2e and systemd-logs plugins from core

### ISSUE THREAD

**Issue by johnSchnake** [vmware-tanzu/sonobuoy#1315]
**Describe the solution you'd like**
We should ideally be able to have the e2e and systemd-logs plugin YAML treated like any other plugin. This is going to help in a few ways:

- simplifying flag/configuration logic
- removing/reducing one-off logic for some legacy e2e/image issues
- hopefully find/add some new features that all plugins can utilize instead of just e2e (e.g. we just made variable images a general feature)

**Anything else you would like to add:**
Longer term this is related to #1292 but before we _fully_ remove it from code we should probably pull it out in a provisional way to ensure we dont break things.

My first guess at breaking it down would be:
 - write the plugin as YAML that is just stored in the codebase so as devs we can run it through the same 'read file' code path
 - feature gate the e2e plugin to always read that YAML so that we can leverage the unit tests/integration tests fully to find the issues via an env var or something (minimal change to the tests/logic elsewhere)
 - A few PRs to reroute the e2e/systemdlogs specific flags to the more general cases. E.g. e2e-focus is just a special case for --plugin-env=e2e.E2E_FOCUS
 - A PR to facilitate adding secrets in some way to plugins just like we did configmaps. This is already in place just for the e2e plugin in order to have ssh keys for certain tests.

See where we land after getting those done.

**Comment by vladimirvivien** [comment]
This is great to see we're moving in that direction. Awesome take on the problem/possible solutions.

**Comment by johnSchnake** [comment]
So far I have extracted logic for the e2e-repo-list, ssh-key/user, and focus/skip/parallel flags (if you count PRs that are up but not merged).

Still todo:
 [ ] applying the `imagePullPolicy` not just to the e2e/systemd plugins but just to all plugins; I think thats an easy approach and pretty easy to defend.

 [ ] update code to handle the `--kube-conformance-image` and `systemd-logs-image` flags. Thats generally easy to; I'll probably make a new flag `--plugin-image=plugin:NEWIMAGE` and these 2 are just a special case of that flag. I dont know how often that would be used, but it solves the problem and adds functionality to other plugins.

 [ ] Add a static YAML in code as discussed above (e.g. adding a logic gate too) and start loading it into the "StaticPlugins" field instead of DynamicPlugins. I dont want to rush into this part because I do want to consider impact on any users. The dynamic plugins allows someone to generate a manifest with just knowing they want the "e2e" plugin, they dont have to know/pass the whole YAML for it. Since we have the methods in code to get the E2E yaml that probably isnt much of an issue. I do think that if we wire it up differently we may end up changing _lots_ of tests though in a way thats very verbose.

**Comment by johnSchnake** [comment]
Going to work on and finish the DD for #1292 before finishing this; I'm at a place where I could implement it in various ways so I want to be sure we're doing this last leg correctly

**Comment by johnSchnake** [comment]
So we've talked about this a bit more and I think we decided a bit more:
 - we wont _completely_ remove e2e/systemd-logs from core code
 - With the implementation of plugin installation, we want to allow installing an `e2e` plugin that would override the built in one. This allows us to ship with the defaults available, but also allows users/us to hotswap the plugin by just running `sonobuoy plugin install ....`

Since we are not going to explicitly remove it we don't even have to do anything else on this I think. The main point was that we can now override the e2e plugin and still have all the same features/flags supported.

Closing.
